gensim/matutils.py

```diff
@@ -479,6 +479,42 @@ def isbow(vec):
         return False
     return True
 
+def symmetric_kl(distrib_p, distrib_q):
+    return numpy.sum([stats.entropy(distrib_p, distrib_q), stats.entropy(distrib_p, distrib_q)])
+
+def arun_metric(min_num_topics=10, max_num_topics=50, iterations=10):
+    """
+    Implements Arun metric to estimate the optimal number of topics:
+    Arun, R., V. Suresh, C. V. Madhavan, and M. N. Murthy
+    On finding the natural number of topics with latent dirichlet allocation: Some observations.
+    In PAKDD (2010), pp. 391–402.
+    
+    Args:
+        min_num_topics: Minimum number of topics to test
+        max_num_topics: Maximum number of topics to test
+        iterations: Number of iterations per value of k
+    
+    Returns: 
+        A list of len (max_num_topics - min_num_topics) with the average symmetric KL divergence for each k
+    
+    Thanks to Adrien Guille for the implementation of the metric.
+    
+    """
+    kl_matrix = []
+    for j in xrange(iterations):
+        kl_list = []
+        l = np.array([sum(corpus.vector_for_document(doc_id)) for doc_id in xrange(corpus.size)])  # document length
+        norm = np.linalg.norm(l)
+        for i in xrange(min_num_topics, max_num_topics + 1):
+            c_m1 = np.linalg.svd(self.topic_word_matrix.todense(), compute_uv=False)
```

Comment:
`self` is undefined, this function isn't a method of any class -> `topic_word_matrix` is undefined too.